src/toxcore/dht/server.rs

```diff
@@ -839,38 +878,39 @@ mod tests {
         let nat_res = NatPingResponse { id: random_u64() };
         let nat_payload = DhtRequestPayload::NatPingResponse(nat_res);
         let dht_req = DhtRequest::new(&precomp, &bob_pk, &bob_pk, nat_payload.clone());
-        let client = alice.create_client(&addr, bob_pk);
-        assert!(alice.handle_nat_ping_resp(client, dht_req, nat_res).wait().is_ok());
+        let mut state = alice.state.write();
+        let client = alice.create_client(&addr, bob_pk, &state.peers_cache);
+        assert!(alice.handle_nat_ping_resp(client, dht_req, nat_res, &state.kbucket).wait().is_ok());
```

Comment:
You should test it by passing special packet, not by calling `handle_nat_ping_resp` directly. See https://github.com/tox-rs/tox/blob/master/src/toxcore/tcp/server/server.rs#L430

Reply:
Test exists which is like above link.
https://github.com/tox-rs/tox/blob/09f1278d6f957bf51bd0ba9628eac0252739e971/src/toxcore/dht/server.rs#L630-L643